src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1317,6 +1348,167 @@ public void Dispose()
                     disposable?.Dispose();
                 }
             }
+
+            public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto(JsonOperationContext context)
+            {
+                var buffer = new byte[_totalSize];
```

Comment:
This is a short lived buffer that may be large. Let's use `ArrayPool` here

Reply:
Issue created
[RavenDB-11852](http://issues.hibernatingrhinos.com/issue/RavenDB-11852)